src/ddmd/irstate.d

```diff
@@ -76,13 +77,19 @@ struct IRState
             deferToObj = irs.deferToObj;
             varsInScope = irs.varsInScope;
             labels = irs.labels;
+            mayThrow = irs.mayThrow;
         }
     }
 
-    this(Module m, FuncDeclaration s)
+    this(Module m, FuncDeclaration fd, Array!(elem*)* varsInScope, Dsymbols* deferToObj, Label*[void*]* labels)
```

Comment:
As far as I can see, you converted the only place that sets the `labels`  and `deferToObj` fields to use this constructor. If you want to improve the encapsulation, you can now mark those fields as `private`, rename them by prepending an underscore and expose them via a getter properties with the original name.